core/src/scheduled_jobs/pending_validations.rs

```diff
@@ -0,0 +1,26 @@
+use crate::{
+    context::Context, network::entry_with_header::EntryWithHeader,
+    workflows::hold_link::hold_link_workflow,
+};
+use holochain_core_types::{cas::content::Address, entry::entry_type::EntryType};
+use std::{sync::Arc, thread};
+
+pub type PendingValidation = Box<(EntryWithHeader, Vec<Address>)>;
```

Comment:
Is the `Vec<Address>` even used anywhere? Might be able to take it out. Especially once we have a function to calculate dependencies, we can just generate them. Also this would get stale if we have to retry validation multiple times, some of the deps might be resolvable, not all, and then we should be updating this list of dependencies.

Reply:
Yeah, this is not used yet. I added it to show how this can be used for the general case as well. I thought we would then run the callback once during validation and then check from the scheduler if those entries are there, before retrying validation. I was also thinking about storing the validation packagage which we now re-request from the source every time.. so lots of room for improvement. I just wanted to get the basic fix in first.

Reply:
not sure about what @maackle is saying here because:

- calculating deps could be a heavy operation so we don't want to redo it and we don't have to because it is pure
- yes the deps might be resolvable but that doesn't mean we get a different array of deps

let's have this conversation in a followup and not derail the MVP

Reply:
Viewing it as a cache makes sense. I was viewing it as a list of dependencies we're still waiting for, and that wouldn't be correct unless we keep this list updated. Makes sense as a cache.